src/App.tsx

```diff
@@ -1,26 +1,244 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import * as todoServices from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { ErrorType } from './types/ErrorType';
+
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(
+    ErrorType.ERROR_DEFAULT,
+  );
+  const [filterTodoBy, setFilterTodoBy] = useState<FilterType>(FilterType.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
why do you need isLoading state? it updates in the same cases as tempTodo state. Looks like enough only tempTodo in cases where you need isLoading state.